boost-install.jam

```diff
@@ -688,21 +761,11 @@ rule boost-install ( VERSION : libraries * )
     install install-detect-toolset : $(boost-install-dir)/BoostDetectToolset.cmake : <location>$(lib-locate)/cmake <name>BoostDetectToolset-$(VERSION).cmake ;
     $(p).mark-target-as-explicit install-detect-toolset ;
 
-    # Gather dependencies
-
-    local deps = ;
-
-    for local lib in $(libraries)
-    {
-        local lname = [ MATCH boost_(.*) : $(lib) ] ;
-        deps += [ deps $(lname) ] ;
-    }
-
-    deps = [ sequence.unique $(deps) ] ;
-
     # Target install
+    generate install-dependencies : $(libraries) : <generating-rule>@boost-install%generate-dependencies <name>install ;
```

Comment:
I suppose that if we use the `boost-install%rulename` syntax in the other places we can avoid exporting the generating rules in Jamroot?